Observer/SalesOrderCreditmemoSaveAfter.php

```diff
@@ -0,0 +1,61 @@
+<?php
+namespace Balancepay\Balancepay\Observer;
+
+use Balancepay\Balancepay\Model\BalancepayRefundFactory;
+use Balancepay\Balancepay\Model\Request\Factory as RequestFactory;
+use Balancepay\Balancepay\Model\ResourceModel\BalancepayCharge\Collection;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+class SalesOrderCreditmemoSaveAfter implements ObserverInterface
+{
+    /**
+     * @var Collection
+     */
+    private $collection;
+
+    /**
+     * @var RequestFactory
+     */
+    private $requestFactory;
+
+    /**
+     * @var BalancepayRefundFactory
+     */
+    private $balancepayRefundFactory;
+
+    public function __construct(
+        Collection $collection,
+        RequestFactory $requestFactory,
+        BalancepayRefundFactory $balancepayRefundFactory
+    ) {
+        $this->collection = $collection;
+        $this->requestFactory = $requestFactory;
+        $this->balancepayRefundFactory = $balancepayRefundFactory;
+    }
+
+    public function execute(Observer $observer)
+    {
+        $creditMemo = $observer->getEvent()->getCreditmemo();
+        $creditMemoId = $creditMemo->getId();
+        $total = $creditMemo->getBaseGrandTotal();
+        $invoiceId = $creditMemo->getInvoiceId();
```

Comment:
throw if no invoice id?

Reply:
Didnt send the request to balance and added a notice message

Reply:
We send a refund request to balance only if there's a Invoice-ID for a Credit-memo, cause we need charge-id to be sent to API.
So, if invoice_id is not available then we will create a credit memo in magento (but will not send refund request to balance), add on Magneto will add a notice message that refund request is not sent to Balance.

Reply:
Can we fail the credit memo creation in such case?
If not, will the merchant be able to manually update the credit memo status?

Reply:
When we don't have charge id, let's cancel the credit memo for now. If possible, let's display a reason why it failed ("No charge id")